src/test/java/io/jenkins/plugins/analysis/warnings/ReferenceFinderITest.java

```diff
@@ -0,0 +1,740 @@
+
+        result = scheduleBuildAndAssertStatus(project, Result.UNSTABLE);
+        assertThat(result).hasTotalSize(6);
+        assertThat(result).hasOverallResult(Result.UNSTABLE);
+
+        project.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(project, "eclipse.txt");
+
+        result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        assertThat(result).hasTotalSize(8);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+    }
+
+    /**
+     * Checks if the reference ignores the result of the last build and therefore returns an unstable in the end.
+     */
+    @Test
+    public void shouldCreateUnstableResultWithNotIgnoredUnstableInBetween() throws IOException, InterruptedException {
+        FreeStyleProject project = createJobWithWorkspaceFile(jobName, "eclipse6Warnings.txt");
+        IssuesRecorder issuesRecorder = enableWarnings(project, publisher -> {
+            publisher.setIgnoreAnalysisResult(true);
+        });
+
+        AnalysisResult result;
+        result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        assertThat(result).hasTotalSize(6);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+        project.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(project, "eclipse4Warnings.txt");
+        issuesRecorder.setUnstableTotalAll(3);
+
+        result = scheduleBuildAndAssertStatus(project, Result.UNSTABLE);
+        assertThat(result).hasTotalSize(4);
+        assertThat(result).hasOverallResult(Result.UNSTABLE);
+
+        project.getSomeWorkspace().deleteContents();
+        issuesRecorder.setUnstableNewAll(3);
+        issuesRecorder.setUnstableTotalAll(9);
+        copyFilesToWorkspace(project, "eclipse.txt");
+
+        result = scheduleBuildAndAssertStatus(project, Result.UNSTABLE);
+        assertThat(result).hasTotalSize(8);
+        assertThat(result).hasOverallResult(Result.UNSTABLE);
+    }
+
+    /**
+     * Checks if the reference only looks at complete success builds instead of just looking at the eclipse result.
+     * Should return an unstable result.
+     */
+    @Test
+    public void shouldCreateUnstableResultWithOverAllMustBeSuccess() throws IOException, InterruptedException {
+        FreeStyleProject project = createJobWithWorkspaceFile(jobName, "eclipse2Warnings.txt");
+        IssuesRecorder issuesRecorder = enableWarnings(project, publisher -> {
+            publisher.setOverallResultMustBeSuccess(true);
+            publisher.setEnabledForFailure(true);
+            publisher.setUnstableNewAll(3);
+        });
+        AnalysisResult result;
+
+        result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+        project.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(project, "eclipse4Warnings.txt", "checkstyle.xml");
+        Shell shell = new Shell("exit 1");
```

Comment:
Die 2 Zeilen kommen auch oft vor: auslagern in Methode `addFailingStep(project)`

Reply:
Weiter unten muss ich die Shell wieder entfernen. Macht das Sinn für 3 Zeilen 2 Methoden zu schreiben?